src/Zynga/Framework/IO/Disk/V1/Manager.hh

```diff
@@ -167,14 +167,17 @@ class Manager implements DiskIOManagerInterface {
 
     if (strpos($in, '\'') !== false ||
         strpos($out, '\'') !== false) {
-      throw new InvalidFileNameException($out);
+      throw new InvalidFileNameException("in=$in, out=$out");
     }
 
     if (file_exists($out)) {
       throw new InvalidFileNameException($out);
     }
 
-    exec("tar --absolute-names -cvf '$out' '$in' 2>/dev/null");
+    $inDir = $this->directoryName($in);
+    $trimmedIn = ltrim(substr($in, strlen($inDir)), '/');
+    $exec = "tar -C '$inDir' -cvf '$out' '$trimmedIn' 2>/dev/null";
+    exec($exec);
```

Comment:
We should use the return code (and potentially returned output) of exec here to throw an exception if return code is non-zero.